### PR TITLE
added the wiki for contract deployment and instantiation

### PR DESCRIPTION
**Pull request by bahdotsh**
## 1. Overview

Added the wiki for contract deployment and instatntiation

### CODE REVIEW

File: contrib/local/CONTRACTS.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Deployment and Instantiation
```
Comment (Anmol1696) [inline contrib/local/CONTRACTS.md:1]: can you please move the MD to `docs/resources` dir and call it `TestnetContract.md`

File: contrib/local/CONTRACTS.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  --instantiate-everybody "true" \
+  --keyring-backend test \
+  --from $VAL_ADDR --gas auto --fees 10000stake -y \
+  --chain-id $CHAIN_ID \
+  -b block -o json --gas-adjustment 1.5)
+  
+echo $RESP 
+```
+
+Now resp has the proposalID, extract the proposal_ID and vote on it
```
Comment (Anmol1696) [inline contrib/local/CONTRACTS.md:31]: now response

File: contrib/local/CONTRACTS.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+PROPOSAL_ID=$(echo "$RESP" | jq -r '.logs[0].events[] | select(.type == "submit_proposal") | .attributes[] | select(.key == "proposal_id") | .value')
+
+persistenceCore tx gov vote $PROPOSAL_ID yes --from $VAL_ADDR --yes --chain-id $CHAIN_ID \
+    --fees 500stake --gas auto --gas-adjustment 1.5 -b block --keyring-backend test -o json | jq
+```
+
+Now get the $CODE_ID by doing `wasm list-code` after the proposal passes and can proceed towards instantiation
+
+```
+persistenceCore tx wasm instantiate "$CODE_ID" "$INIT" --admin="$(persistenceCore keys show $VAL_ADDR -a --keyring-backend test)" \
```
Comment (Anmol1696) [inline contrib/local/CONTRACTS.md:43]: can you remove this.. we wont have wasm instantiate as a tx but a proposal itself

File: contrib/local/CONTRACTS.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  --from $VAL_ADDR --amount "10000stake" --label "local0.1.0" --gas-adjustment 1.5 --fees "10000stake" \
```
Comment (Anmol1696) [inline contrib/local/CONTRACTS.md:44]: lets make all `--gas-adjustment 1.1` and also change all `stake` denom to `uxprt`

File: contrib/local/CONTRACTS.md (hunk 3 above)
Comment (Anmol1696) [reply contrib/local/CONTRACTS.md:43]: can add the gov proposal voting step wait as well.
* submit instantiate proposal
* vote on the proposal, wait for it to pass
* submit proposal for instantiate
* vote on the proposal, wait for it to pass

Then interact with the proposal.

File: contrib/local/CONTRACTS.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+You can set the `node` to the persistenceCore config and don't have to worry about passing that flag always
+
+```
+persistenceCore config node https://rpc.testnet.persistence.one:443
+```
+
+To upload the contract
+
+```
+RESP=$(persistenceCore tx gov submit-proposal wasm-store "path/to/the/compiled/wasm" \
```
Comment (Anmol1696) [inline contrib/local/CONTRACTS.md:17]: replace `"path/to/the/compiled/wasm"` to `"<path/to/the/compiled/wasm>"` so we know we have to replace it

File: contrib/local/CONTRACTS.md (hunk 3 above)
Comment (Anmol1696) [reply contrib/local/CONTRACTS.md:43]: also in the start of the doc, in the overview can mention the steps that will be taking place... maybe helpful

**Review by Anmol1696 — CHANGES_REQUESTED** [review]

File: docs/resources/TestnetContracts.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+To upload the contract
+
+```
+RESP=$(persistenceCore tx gov submit-proposal wasm-store "<path/to/the/compiled/wasm>" \
+  --title "title" \
+  --description "description" \
+  --deposit 10000uxprt \
+  --run-as $VAL_ADDR \
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:29]: what is VAL addr, maybe we dont need this...
change the name to test-core-1 address
also chain id need to be defined.

Could you please review the PR urself.. it should all make sense, give it a through read

Comment (bahdotsh) [reply docs/resources/TestnetContracts.md:29]: $VAL_ADDR is the address of the validator. We have done it like this for all the contracts

Comment (Anmol1696) [reply docs/resources/TestnetContracts.md:29]: ya as a user this done be the variable we will be using.. and the address wont be a validator address

Comment (Anmol1696) [reply docs/resources/TestnetContracts.md:29]: change this to something more generic and mention what this value should be, basically an address with some test uxprt in there.
I dont mind if we use variables but mention how and what to set them

**Review by Anmol1696 — CHANGES_REQUESTED** [review]

File: docs/resources/TestnetContracts.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+To upload the contract via proposal
+
+```
+RESP=$(persistenceCore tx gov submit-proposal wasm-store "<path/to/the/compiled/wasm>" \
+  --title "title" \
+  --description "description" \
+  --deposit 10000uxprt \
+  --run-as $TEST_KEY \
+  --instantiate-everybody "true" \
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:30]: set this to nobody please. 
Default it nobody

File: docs/resources/TestnetContracts.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+- Vote on the proposal, wait for it to pass
+- The contract is now instantiated and ready to be interacted with.
+
+See the list of codes that was uploaded to the testnet previously.
+```
+persistenceCore query wasm list-code --node https://rpc.testnet.persistence.one:443
+```
+
+You can set the `node` to the persistenceCore config and don't have to worry about passing that flag always
+
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:17]: no need to new line

File: docs/resources/TestnetContracts.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+```
+persistenceCore config node https://rpc.testnet.persistence.one:443
+```
+
+To upload the contract via proposal
+
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:23]: remove new line

File: docs/resources/TestnetContracts.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+  --from $TEST_KEY --gas auto --fees 10000uxprt -y \
+  --chain-id test-core-1 \
+  -b block -o json --gas-adjustment 1.1)
+  
+echo $RESP 
+```
+The `$TEST_KEY` can be any valid persistenceAddress. Make sure it has some test tokens.
+
+Now `$RESP` has the proposalID, extract the proposal_ID and vote on it
+
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:41]: remove new line

File: docs/resources/TestnetContracts.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+```
+PROPOSAL_ID=$(echo "$RESP" | jq -r '.logs[0].events[] | select(.type == "submit_proposal") | .attributes[] | select(.key == "proposal_id") | .value')
+
+persistenceCore tx gov vote $PROPOSAL_ID yes --from $TEST_KEY --yes --chain-id test-core-1 \
+    --fees 500uxprt --gas auto --gas-adjustment 1.1 -b block --keyring-backend test -o json | jq
+```
+
+The contract is instantiated via a gov-proposal 
+
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:50]: remove new line

File: docs/resources/TestnetContracts.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+Get the $CODE_ID from `wasm list-code` after the proposal passes and proceed towards instantiation
+
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:52]: remove new line

File: docs/resources/TestnetContracts.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+    "address": "$TEST_KEY",
+    "amount": "123456789000"
+  }]
+}
+EOF
+)
+
+```
+
+
```
Comment (Anmol1696) [inline docs/resources/TestnetContracts.md:88]: extra new lines

**Review by Anmol1696 — CHANGES_REQUESTED** [review]

**Review by Anmol1696 — APPROVED** [review]
LGTM

**Pull request merged**